Tests/ApolloCodegenTests/CodeGeneration/FileGenerators/EnumFileGeneratorTests.swift

```diff
@@ -0,0 +1,64 @@
+import XCTest
+import Nimble
+@testable import ApolloCodegenLib
+import ApolloCodegenTestSupport
+
+class EnumFileGeneratorTests: XCTestCase {
+  override func tearDown() {
+    CodegenTestHelper.deleteExistingOutputFolder()
+
+    super.tearDown()
+  }
+
+  func test_generate_givenSchemaType_shouldOutputToPath() throws {
+    // given
+    let schema = """
+    type Query {
+      books: [Book!]!
+    }
+
+    type Book {
+      name: String!
+      rating: Rating!
+    }
+
+    enum Rating {
+      GOOD
+      BAD
+    }
+    """
+
+    let operation = """
+    query getBooks {
+      books {
+        name
+        rating
+      }
+    }
+    """
+
+    let ir = try IR.mock(schema: schema, document: operation)
+    let ratingType = try ir.schema[enum: "Rating"].xctUnwrapped()
+
+    let rootURL = URL(fileURLWithPath: CodegenTestHelper.outputFolderURL().path)
+    let fileURL = rootURL.appendingPathComponent("Rating.swift")
+
+    let mockFileManager = MockFileManager(strict: false)
+
+    mockFileManager.mock(closure: .createFile({ path, data, attributes in
+      expect(path).to(equal(fileURL.path))
+      expect(String(data: try! data.xctUnwrapped(), encoding: .utf8))
```

Comment:
We don't need to be testing the actual data in these tests. If the test is to ensure that we are generating output to the right path, then just test that.

Every time we change we the rendered output shape looks like, this test shouldn't fail and have to be changed.

Tests for the actual generated content are in the unit tests for the templates. The `EnumFileGenerator` and all other file generators should really be calling into templates that are located in the "Templates" folder, like I have for `SchemaTemplate` and the others.

Reply:
I've reduced the test area for all these tests in https://github.com/apollographql/apollo-ios/pull/2093/commits/894646b4ea6b550be952c7618c261bf78a3971ef.